app/backend/app.py

```diff
@@ -307,6 +309,14 @@ async def setup_clients():
         )
     elif OPENAI_HOST == "local":
         openai_client = AsyncOpenAI(base_url=os.environ["OPENAI_BASE_URL"], api_key="no-key-required")
+    elif OPENAI_HOST == "azure_custom":
+        token_provider = get_bearer_token_provider(azure_credential, "https://cognitiveservices.azure.com/.default")
+        # Store on app.config for later use inside requests
+        openai_client = AsyncAzureOpenAI(
+            api_version="2023-07-01-preview",
+            azure_endpoint=os.environ["OPENAI_BASE_URL"],
```

Comment:
The only difference in this code is the endpoint, right? In that case, I think it'd be preferred to change the original if to "startswith("azure")" and then alter the endpoint only inside that block. Or are you thinking there'll future differentiation between the setup?